src/coreclr/vm/interoputil.cpp

```diff
@@ -790,21 +790,11 @@ BOOL IsComObjectClass(TypeHandle type)
     CONTRACTL_END;
 
 #ifdef FEATURE_COMINTEROP
-    if (!type.IsTypeDesc())
-    {
-        MethodTable *pMT = type.AsMethodTable();
-
-        if (pMT->IsComObjectType())
-        {
-            // May be __ComObject or typed RCW. __ComObject must have already been loaded
-            // if we see an MT marked like this so calling the *NoInit method is sufficient.
-
-            return pMT == g_pBaseCOMObject;
-        }
-    }
-#endif
-
+    // May be __ComObject or typed RCW.
+    return type == TypeHandle(g_pBaseCOMObject);
```

Comment:
We should create a test for getting the `GUID` from the CoClass scenario. See test example [here](https://github.com/dotnet/runtime/blob/81efcade9e543a2c03457b3bc87a2bc76b36057e/src/tests/Interop/COM/NETClients/Primitives/NumericTests.cs#L21).

```csharp
var server = new ComServer();
Console.WriteLine(server.GetType().IsCOMObject);
Console.WriteLine(server.GetType().GUID);

[ComImport]
[Guid("...")]
class ComServer;
```

Reply:
I have pushed a change to add Assert for Type.GUID to the existing test for this scenario. (The existing test is for Type.Guid alias on Marshal.)